perfkitbenchmarker/vpn_service.py

```diff
@@ -0,0 +1,344 @@
+# Copyright 2020 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""VPN support for network benchmarks.
+
+This module contains the main VPNService class, which manages the VPN lifecycle,
+the VPN class, which manages the VPN tunnel lifecycle between two endpoints, and
+the TunnelConfig class, which maintains the parameters needed to configure a
+tunnel between two endpoints. Related: perfkitbenchmarker.network
+module includes the BaseVpnGateway class to manage VPN gateway endpoints.
+"""
+
+import itertools
+import json
+import logging
+import re
+import threading
+import uuid
+from enum import Enum
+
+from perfkitbenchmarker import context
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import resource
+from perfkitbenchmarker import vm_util
+
+flags.DEFINE_integer('vpn_service_tunnel_count', None,
+                     'Number of tunnels to create for each VPN Gateway pair.')
+flags.DEFINE_integer('vpn_service_gateway_count', None,
+                     'Number of VPN Gateways to create for each vm_group.')
+flags.DEFINE_string('vpn_service_name', None,
+                    'If set, use this name for VPN Service.')
+flags.DEFINE_string('vpn_service_shared_key', None,
+                    'If set, use this PSK for VPNs.')
+flags.DEFINE_string('vpn_service_routing_type', None, 'static or dynamic(BGP)')
```

Comment:
Can you make this a `flags.DEFINE_enum`?

Reply:
Fixed.